### DISCUSSION TITLE
Klick'r 3.0.0-beta05

### DISCUSSION THREAD

**Discussion by Nain57** [Nain57/Smart-AutoClicker#465] [Announcements]
> [!IMPORTANT]
> **This is a beta build.**
> It might be unstable so please backup your scenarios before updating to this version. It will be only availble on direct download here, and not on PlayStore/FDroid.

Sorry for the update delay, I have been very busy with my personal life lately. Last beta have shown very stable according to crashlytics, and i'm very happy to introduce you to the final changes before the production release.

## Unify Ui between modes

Ui for the regular scenarios have been praised by many users, while the Ui for the Smart Scenarios was going more and more a "menu hell". So i have applied the same UX to the Actions and Image Conditions list, allowing you to easily interact and test while creating them.

<p align="center">
<img src="https://github.com/user-attachments/assets/10963a2c-6964-46bf-809a-8c0b37945cc8" height="400"> <img src="https://github.com/user-attachments/assets/e72e70a5-a8b8-4203-ae5a-b41dfe3502ff" height="400"> <img src="https://github.com/user-attachments/assets/f7b403b3-6756-442b-934f-ec1c0872f3c3" height="400"> 
</p>

This new UI will allow you to easily understand the content of an Action/Conditions, as well as providing other new features in a centralized way. You can test the focused item in one click to verify if it works as intented, allowing you to save a lot of time during the scenario creation.
Also, reordering actions was a bit tedious when you you have a lot of them. You can now move them using the '<<' and '>>' buttons, or select directly the new position by click in the displayed index (1/2). 

All of the new Smart Scenario features have also been backported to the regular Scenario.

## Click/Swipe capture

A new option have been added to the Actions menu for Regular and Smart Scenarios, the gesture capture. It allows you to capture a click or a swipe, with the positions and duration automatically filled for you.

<p align="center">
<img src="https://github.com/user-attachments/assets/d09ee1b6-e541-4850-bcc1-9647112d158f" height="400">
</p>

This will allows you to quickly create an action, and combined with the test action button, you can test it immediately.

> [!NOTE] 
> This is as close as possible to "Record & Play" feature. 
> I have tested it on a lot of apps that propose it, and I was never really convinced. The issue with Android, you can't register touch events and let them pass though. You need to either capture first and replay, or do some weird blinking between capture a letting through to execute the user action. 
> Both of those solutions have a lot of flaws and I don't think they are a good solution. That's why I have choosen to propose something different, capture and replay are separated, but available on the same screen in one click.
> My firsts testers find it very efficient, but let me know if you like it.

## New Langage supported: Brazilian Portuguese

Thanks a lot to @Vaz15k for its contribution!

<hr /><em>This discussion was created from the release <a href='https://github.com/Nain57/Smart-AutoClicker/releases/tag/3.0.0-beta05'>Klick'r 3.0.0-beta05</a>.</em>

**Comment by Mohmotte** [comment]
Where is the afk file please ?

**Reply by Nain57** [reply]
Attached to the release.
Publishing the release note trigger the build of the apk, so you usually have to wait around 15 minutes to have it available.

**Comment by noobiessss** [comment]
WOW. great stuff. thanks for your hard work!

**Comment by noobiessss** [comment]
please bring back the action screen event in list format or give us the option to change the view to list. the new format is hard to manage.
![Capture](https://github.com/user-attachments/assets/279b863c-10e0-4f19-9adc-083f105875d1)
